civihr_employee_portal/js/reports.js

```diff
@@ -792,12 +792,43 @@
           }
         });
       })
-      .controller('FiltersController', function (AbsencePeriod) {
-        this.format = 'dd/MM/yyyy';
-        this.placeholderFormat = 'dd/MM/yyyy';
-        this.date = new Date();
-        this.filtersCollapsed = true;
-      })
+      .controller('FiltersController', ['REPORT_NAME', 'AbsencePeriod',
+        function (REPORT_NAME, AbsencePeriod) {
+          var vm = this;
+
+          vm.dateFormat = 'dd/MM/yyyy';
+          vm.filters = {};
+          vm.loading = { dates: false };
+
+          (function init () {
+            if (REPORT_NAME === 'leave_and_absence') {
+              vm.loading.dates = true;
+              initCurrentAbsencePeriodFilterDates().finally(function () {
+                vm.loading.dates = false;
+              });
+            } else {
+              vm.filters.date = new Date();
+            }
+          })();
+
+          /**
+           * Initializes the date filters using the start and end dates for the
+           * current absence period.
+           *
+           * @return {Promise}
+           */
+          function initCurrentAbsencePeriodFilterDates () {
+            return AbsencePeriod.getCurrent().then(function (currentPeriod) {
+              if (!currentPeriod) {
+                return;
```

Comment:
What happens if the current period does not exists? Does it fall back smh?

Reply:
As discussed with @mi-kon they default to blank:

![leave reports hr17 9000 10](https://user-images.githubusercontent.com/1642119/35101198-0bbed608-fc35-11e7-815e-17f6f3a26eed.png)